components/brave_wallet/browser/wallet_data_files_installer.cc

```diff
@@ -0,0 +1,185 @@
+
+bool WalletDataFilesInstallerPolicy::
+    SupportsGroupPolicyEnabledComponentUpdates() const {
+  return false;
+}
+
+bool WalletDataFilesInstallerPolicy::RequiresNetworkEncryption() const {
+  return false;
+}
+
+update_client::CrxInstaller::Result
+WalletDataFilesInstallerPolicy::OnCustomInstall(
+    const base::DictionaryValue& manifest,
+    const base::FilePath& install_dir) {
+  return update_client::CrxInstaller::Result(update_client::InstallError::NONE);
+}
+
+void WalletDataFilesInstallerPolicy::OnCustomUninstall() {}
+
+void WalletDataFilesInstallerPolicy::ComponentReady(
+    const base::Version& version,
+    const base::FilePath& path,
+    std::unique_ptr<base::DictionaryValue> manifest) {
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock(), base::TaskPriority::BEST_EFFORT},
+      base::BindOnce(&WalletDataFilesInstallerPolicy::TokenListReady,
+                     base::Unretained(this), path, std::move(manifest)),
+      base::BindOnce(&WalletDataFilesInstallerPolicy::UpdateTokenRegistry,
+                     base::Unretained(this)));
+}
+
+bool WalletDataFilesInstallerPolicy::VerifyInstallation(
+    const base::DictionaryValue& manifest,
+    const base::FilePath& install_dir) const {
+  return true;
+}
+
+// The base directory on Windows looks like:
+// <profile>\AppData\Local\Google\Chrome\User Data\BraveWallet\.
+base::FilePath WalletDataFilesInstallerPolicy::GetRelativeInstallDir() const {
+  return base::FilePath::FromUTF8Unsafe(kWalletBaseDirectory);
+}
+
+void WalletDataFilesInstallerPolicy::GetHash(std::vector<uint8_t>* hash) const {
+  hash->assign(kWalletDataFilesSha2Hash,
+               kWalletDataFilesSha2Hash + base::size(kWalletDataFilesSha2Hash));
+}
+
+std::string WalletDataFilesInstallerPolicy::GetName() const {
+  return kWalletDataFilesDisplayName;
+}
+
+update_client::InstallerAttributes
+WalletDataFilesInstallerPolicy::GetInstallerAttributes() const {
+  return update_client::InstallerAttributes();
+}
+
+std::vector<mojom::ERCTokenPtr> WalletDataFilesInstallerPolicy::TokenListReady(
+    const base::FilePath& install_dir,
+    std::unique_ptr<base::DictionaryValue> manifest) {
+  std::vector<mojom::ERCTokenPtr> erc_tokens;
+  // On some platforms (e.g. Mac) we use symlinks for paths. Convert paths to
+  // absolute paths to avoid unexpected failure. base::MakeAbsoluteFilePath()
+  // requires IO so it can only be done in this function.
+  const base::FilePath absolute_install_dir =
+      base::MakeAbsoluteFilePath(install_dir);
+
+  if (absolute_install_dir.empty()) {
+    LOG(ERROR) << "Failed to get absolute install path.";
+    return erc_tokens;
+  }
+
+  const base::FilePath token_list_json_path =
+      absolute_install_dir.Append("contract-map.json");
+  std::string token_list_json;
+  if (!base::ReadFileToString(token_list_json_path, &token_list_json)) {
+    LOG(ERROR) << "Can't read token list file.";
+  }
+
+  ParseTokenList(token_list_json, &erc_tokens);
```

Comment:
What happens if this return false?

Reply:
return here is intentionally ignored so that the vector is just empty

Reply:
Updated handling here 03cfad13253c57cdb043529378d80f08129f2c39